server/events/event_parser.go

```diff
@@ -34,6 +35,8 @@ import (
 const gitlabPullOpened = "opened"
 const usagesCols = 90
 
+var lastBitbucketSha, _ = lru.New[string, string](300)
```

Comment:
Can we do this without adding a new dependency? There are so many dependencies in atlantis that it's difficult to maintain them and keep them up to date. Each new dependency also adds to the blast radius and increases the build time of each build so we generally want to be very selective when adding new dependencies.

Reply:
I could do a simple dict, but the downside would be ever-increasing memory usage. I'm not sure what the exact rate is, as it would depend on go internals and how often PR's happen. If people make 20 PR's a day then maybe that's ~50*20=1KB daily? We can assume there's annual patching that would involve a restart of the server, so I suppose the memory leak wouldn't be that bad. If you feel that's preferable I can change it.

However, I should note that https://github.com/hashicorp/golang-lru:
* Is a very small dependency
* popular
* has a small API
* Is under the hashicorp org

So as far as dependencies go this is about as light and low-risk as you can get.